setup_utils.py

```diff
@@ -35,15 +35,6 @@ def pkg_config_read(library: str, var: str) -> str:
     return path
 
 
-def in_virtualenv() -> bool:
```

Comment:
Let me know what needs to be added to the commit message:
```
  commit 5b0d2fa1bf714d7a952a27a59f1ce070159ff352
  Author: Brett Holman <brett.holman@canonical.com>
  Date:   Fri May 31 12:56:50 2024 -0600
  
      ci(mypy): Set default follow_imports value (#5350)
      
      Remove code for detecting a virtualenv. This has been broken since ~3.7.
  
   pyproject.toml |  1 -
   setup.py       | 17 +++++++----------
   setup_utils.py |  9 ---------
   3 files changed, 7 insertions(+), 20 deletions(-)
```

Reply:
Mypy correctly points out that this isn't used anywhere on 3.12. Checking for the presence of a virtualenv hasn't worked since Python 3.7, and since nobody has bothered to make it work on >3.7, the assumption is that nobody cares about it working on 3.7 either. This appears to be a relic of the past.

Other factors: 

- We planned to drop support for 3.7 a year ago. We held out for a few stragglers that still wanted support, but I think it is time for 3.7 to go.
- Lots of disruption is coming in the packaging space in the upcoming cycle for PEP517. This is just packaging code. If someone still really wants this, they can maintain it themselves.

Aside: Earlier this year when I was trying to work on PEP517 support in my spare time, this code was actually a stumbling block due to the added complexity. Culling unnecessary junk prior to a big refactor or change in functionality helps a lot in the long run to understand what we actually need.

Reply:
> Mypy correctly points out that this isn't used anywhere on 3.12

Didn't realize that. I don't disagree with anything you said. I knew this needed to go; I just didn't see the relation to this PR.

> 3.7

Let's start with 3.6 :wink: 

According to our original proposed calendar, we should still be supporting 3.7 until the end of the month.